### PR TITLE
Only apply datatable plugin if there is a valid table header present

### PR DESCRIPTION
**Pull request by jbarnicle**

### CODE REVIEW

File: regulations/static/regulations/js/source/views/main/main-view.js [hunk 1]
```diff
@@ -255,15 +255,19 @@ var MainView = Backbone.View.extend({
         if (this.dataTables) {
             this.dataTables.destroy();
         }
+        // Only apply the datatables plugin if there is a table header present
         if (this.$el.find('table').length) {
-            this.dataTables = this.$el.find('table').DataTable({
+            this.dataTables = this.$el.find('table').filter(
+                function() {
+                    return $('thead', this).children().length > 0;
+                }).DataTable({
```
Comment (cmc333333) [inline regulations/static/regulations/js/source/views/main/main-view.js]: Would this be clearer?
```javascript
this.dataTables = this.$el.find('table').has('thead').DataTable({
```

Comment (jbarnicle) [inline regulations/static/regulations/js/source/views/main/main-view.js]: The table in question has an empty thead (<thead></thead>) so that won't work

Comment (jbarnicle) [inline regulations/static/regulations/js/source/views/main/main-view.js]: "<thead></thead>"

Comment (jbarnicle) [inline regulations/static/regulations/js/source/views/main/main-view.js]: Not very familiar with has - would .has('thead tr') work?

**Comment by cmc333333** [comment]
Thanks for the rapid response!

**Pull request merged**